### PR TITLE
feat: Adding IAM credential type for AWS

### PR DESCRIPTION
**Pull request by fkautz**
This adds the ability to use IAM credentials in EC2 automatically.

### CODE REVIEW

**Review by jkjell — DISMISSED** [review]

**Review by kairoaraujo — COMMENTED** [review]
Hi @fkautz, thank you for the PR.

I have a minor change request.
 Could you include `ARCHIVISTA_BLOB_STORE_CREDENTIAL_TYPE ` on the `README.md`, [configuration section](https://github.com/in-toto/archivista#configuration)?

**Comment by fkautz** [comment]
Added the data to the readme. Also, I ran into an issue where v6 of the library doesn't work on all AMIs based upon which metadata service EC2 loads up for it. The latest v7 fixes the problem.

File: README.md [hunk 1]
```diff
@@ -45,22 +45,23 @@ default with this docker compose file.
...
+| ARCHIVISTA_LOG_LEVEL                       | INFO                         | Log level. Options are DEBUG, INFO, WARN, ERROR                                               |
+| ARCHIVISTA_CORS_ALLOW_ORIGINS              |                              | Comma separated list of origins to allow CORS requests from                                   |
+| ARCHIVISTA_SQL_STORE_CONNECTION_STRING     | root:example@tcp(db)/testify | SQL store connection string                                                                   |
+| ARCHIVISTA_STORAGE_BACKEND                 |                              | Backend to use for attestation storage. Options are FILE, BLOB, or empty string for disabled. |
+| ARCHIVISTA_FILE_SERVE_ON                   |                              | What address to serve files on. Only valid when using FILE storage backend.                   |
+| ARCHIVISTA_FILE_DIR                        | /tmp/archivist/              | Directory to store and serve files. Only valid when using FILE storage backend.               |
```
Comment (kairoaraujo) [inline README.md:56]:
```suggestion
| ARCHIVISTA_FILE_DIR                        | /tmp/archivista/              | Directory to store and serve files. Only valid when using FILE storage backend.               |
```

**Review by kairoaraujo — COMMENTED** [review]
Nit: as I'm refactoring the tests
We could add it to the config_test.go as well.

**Review by kairoaraujo — DISMISSED** [review]

**Comment by kairoaraujo** [comment]
@fkautz, could you please rebase and resolve the conflict so we can merge it?

**Review by kairoaraujo — APPROVED** [review]

**Pull request merged**